### DISCUSSION TITLE
Regarding Performance <training_time >:  in Automatic Self_Checkout

### DISCUSSION THREAD

**Discussion by Abhijeet241093** [openvinotoolkit/openvino_notebooks#1801] [AI Solution Kits]
Hi @riacheruvu, 

Thank you for your attention. 

**Describe the bug**
In function def get_iou, def intersecting_bboxes, run during training, considerably take running time ?  I means if yolo model working on detection in that case, weather this function  get_iou, & intersecting_bboxes run independently ? Is it slow down yolo model ? performance ? 

Can function get_iou, & intersecting_bboxes participate in training time ?  

 **Expected behavior**
  If function get_iou, & intersecting_bboxes participate in training time ?  In that case, Is their way to separate this functions and yolo detection model, so both run independently ?  so training become faster while adding new label data in custom dataset next time. 

**Screenshots**

 **Installation instructions (Please mark the checkbox)**
[ Yes ] I followed the installation guide at https://github.com/openvinotoolkit/openvino_notebooks#-installation-guide to install the notebooks. 

** Environment information **
Please run `python check_install.py` in the _openvino_notebooks_ directory. If the output is _NOT OK_ for any of the checks, please follow the instructions to fix that. If that does not work, or if you still encounter the issue, please paste the output of check_install.py here.
 
**Additional context**

**Comment by riacheruvu** [comment]
Hi @Abhijeet241093, the `get_iou()` and `intersecting_bboxes()` functions are reserved for inference - they aren't applicable at training time. The use of the functions during inference can increase the runtime of the application by a bit, just because we're doing additional post-processing and cross-checking of the YOLO model's outputs. I hope this answers your question. This discussion is more suited for the [Discussions ](https://github.com/openvinotoolkit/openvino_notebooks/discussions), as it's not specifically a bug - I would encourage you to create a new thread there and we can talk further if you have additional questions. Thanks!

**Comment by riacheruvu** [comment]
Requesting @adrianboguszewski to close this thread, as we can move this thread to the Discussions page. Thank you!

**Reply by adrianboguszewski** [reply]
Converted to discussion
